modules/tournament/src/main/Tournament.scala

```diff
@@ -102,27 +111,135 @@ case class Tournament(
+      val firstLastBonus: Int = math.min(clock.limitSeconds / 3, 120)
+      // take time from first variant and give to last
+
+      times.map(t =>
+        t.take(1).map(v => v - firstLastBonus) ::: t.drop(1).take(t.size - 2) :::
+          t.drop(t.size - 1).map(v => v + extra + firstLastBonus)
+      )
+    } else {
+      medleyNumIntervals.map(List.fill(_)(medleyMinutes.getOrElse(0) * 60))
+    }
+
+  def medleySpeedFactors: Option[List[Double]] =
+    medleyVariantsInTournament.map(_.map(v => medleySpeedChoice.get(v.key).getOrElse(1.0)))
+
+  def medleySpeedChoice: Map[String, Double] = medleyVariants.fold(medleyVariantSpeeds)(v =>
+    v match {
+      case x if isMedleyChessShieldStyle(x)    => medleyChessShieldSpeeds
+      case x if isMedleyDraughtsShieldStyle(x) => medleyDraughtShieldSpeeds
+      case _                                   => medleyVariantSpeeds
+    }
+  )
+
+  def isMedleyChessShieldStyle(variants: List[Variant]): Boolean =
+    id == "shieldChessMedley" || variants.forall(v => medleyChessShieldSpeeds.keys.exists(v.==))
+
+  def isMedleyDraughtsShieldStyle(variants: List[Variant]): Boolean =
+    id == "shieldDraughtsMedley" || variants.forall(v => medleyDraughtShieldSpeeds.keys.exists(v.==))
+
+  def medleyVariantSpeeds: Map[String, Double] = {
+    val slow    = 1.25
+    val medium  = 1.1
+    val quick   = 0.9
+    val fastest = 0.75
+    Map(
+      "standard"      -> medium,
+      "chess960"      -> medium,
+      "kingOfTheHill" -> quick,
+      "threeCheck"    -> fastest,
+      "fiveCheck"     -> quick,
+      "antichess"     -> quick,
+      "atomic"        -> fastest,
+      "horde"         -> slow,
+      "racingKings"   -> fastest,
+      "crazyhouse"    -> quick,
+      "noCastling"    -> medium,
+      "linesOfAction" -> fastest,
+      "scrambledEggs" -> fastest,
+      "frisian"       -> medium,
+      "frysk"         -> quick,
+      "international" -> slow,
+      "antidraughts"  -> slow,
+      "breakthrough"  -> slow,
+      "russian"       -> medium,
+      "brazilian"     -> medium,
+      "pool"          -> medium,
+      "portuguese"    -> medium,
+      "english"       -> medium,
+      "shogi"         -> slow,
+      "xiangqi"       -> medium,
+      "minishogi"     -> fastest,
+      "minixiangqi"   -> quick,
+      "flipello"      -> medium,
+      "flipello10"    -> slow,
+      "oware"         -> slow
+    )
+  }
+
+  def medleyChessShieldSpeeds: Map[String, Double] = {
+    val slow    = 1.25
+    val medium  = 1
+    val fastest = 0.75
+    Map(
+      "kingOfTheHill" -> medium,
+      "threeCheck"    -> fastest,
+      "antichess"     -> medium,
+      "atomic"        -> fastest,
+      "horde"         -> slow,
+      "racingKings"   -> fastest,
+      "crazyhouse"    -> medium
+    )
+  }
+
+  def medleyDraughtShieldSpeeds: Map[String, Double] = {
+    val slow   = 1.25
+    val medium = 1
+    val quick  = 0.75
```

Comment:
why is this quick for Draughts and fastest for chess?

Reply:
changed both to quick.